src/odemis/acq/stitching/_registrar.py

```diff
@@ -746,4 +746,8 @@ def _assemble_mosaic(self):
             if i not in positions:
                 positions[i] = [None, None]
 
-        return numpy.array(list(positions.values())).reshape([num_rows, num_cols, 2])
+        # Sort positions by key
+        positions = numpy.array(sorted(positions.items(), key=lambda x: x[0]))
```

Comment:
You don't need to store it as a numpy array here.

Reply:
indeed, it was actually also failing on python3 (it didn't like converting a mulitdimensional list to a numpy.array)